docs/guides/plugins.md

```diff
@@ -1,59 +1,324 @@
--------------------------------
-It's time to give the rest of the world the opportunity to be awed by your genius. When your plugin is loaded, it needs to let Video.js know this amazing new functionality is now available:
+> **Note:** The value of `this` in the plugin function is the player instance; so, you have access to [its complete API][api-player].
+
+### Register a Basic Plugin
+
+Now that we have a function that does something with a player, all that's left is to register the plugin with Video.js:
 
 ```js
-    videojs.plugin('examplePlugin', examplePlugin);
+videojs.registerPlugin('examplePlugin', examplePlugin);
 ```
 
-From this point on, your plugin will be added to the Video.js prototype and will show up as a property on every instance created. Make sure you choose a unique name that doesn't clash with any of the properties already in Video.js. Which leads us to...
+The only stipulation with the name of the plugin is that it cannot conflict with any existing player method. After that, any player will automatically have an `examplePlugin` method on its prototype!
+
+## Writing a Class-Based Plugin
+
+As of Video.js 6, there is an additional type of plugin supported: class-based plugins.
+
+At any time, you may want to refer to the [Plugin API docs][api-plugin] for more detail.
+
+### Write a JavaScript Class/Constructor
+
+If you're familiar with creating [components](components.md), this process is similar. A class-based plugin starts with a JavaScript class (a.k.a. a constructor function).
+
+This can be achieved with ES6 classes:
+
+```js
+const Plugin = videojs.getPlugin('plugin');
+
+class ExamplePlugin extends Plugin {
+
+  constructor(player, options) {
+    super(player);
+
+    player.addClass(options.customClass);
+    player.on('playing', function() {
+      videojs.log('playback began!');
+    });
+  }
+}
+```
 
-Step 3: Using A Plugin
-----------------------
-There are two ways to initialize a plugin. If you're creating your video tag dynamically, you can specify the plugins you'd like to initialize with it and any options you want to pass to them:
+Or with ES5:
 
 ```js
-    videojs('vidId', {
-      plugins: {
-        examplePlugin: {
-          exampleOption: true
-        }
-      }
+var Plugin = videojs.getPlugin('plugin');
+
+var ExamplePlugin = videojs.extend(Plugin, {
+
+  constructor: function(player, options) {
+    Plugin.call(this, player, options);
+
+    player.addClass(options.customClass);
+    player.on('playing', function() {
+      videojs.log('playback began!');
     });
+  }
+});
+```
+
+For now, this example class-based plugin does the exact same thing as the basic plugin described above - not to worry, we will make it more interesting as we continue!
+
+### Register a Class-Based Plugin
+
+The registration process for class-based plugins is identical to [the process for basic plugins](#register-a-basic-plugin).
+
+```js
+videojs.registerPlugin('examplePlugin', ExamplePlugin);
```

Comment:
Sure, I'm adding a note after this code block that "classes" are still functions.